internal/cli/config/config.go

```diff
@@ -0,0 +1,144 @@
+package config
+
+import (
+	"os"
+	"path"
+
+	"github.com/ghodss/yaml"
+	"github.com/mitchellh/go-homedir"
```

Comment:
I couldn't tell if this library follows the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) where `~/.config/kargo` is preferred over `~/.kargo`. Looking at the source, I don't think it does.

If it doesn't honor `XDG_CONFIG_HOME` we should not use this.

Reply:
Wow... I had not known about this specification. I'll read up on it and make sure we're playing nice with it.

Thanks for this catch!

Reply:
@jessesuen , I read the spec and also looked for a library for Go that might make short work of this. [adrg/xdg](https://github.com/adrg/xdg) looks to be the most popular. What I especially like about this library is that when you do `xdg.ConfigFile(<some relative path>)` it will ensure the directory exists if it doesn't already, which really cuts down on the amount of code _I_ have to write... it's the little things.

I did notice something peculiar, however, that I want to run by you...

On a Mac, if `XDG_CONFIG_HOME` is not defined, this lib defaults config home to `~/Library/Application Support`. 🤔  While this is not what the spec says, this does seem to be the popular opinion about where user-specific config files actually _belong_ on a Mac. In fact, Go's own std lib seems to endorse this opinion, [as is evident in the `os` package](https://pkg.go.dev/os#UserConfigDir).

Also interesting: _On a Mac_, Go's `os.UserConfigDir()` function doesn't honor the `XDG_CONFIG_HOME` env var _at all_, although the `adrg/xdg` does. (I'm beginning to question if this spec was meant to be applicable to Mac and Windows at all. It comes from freedesktop.org so it makes sense that it's implicitly _not_ focused on Mac and Windows.)

Bottom line is that if you were expecting config to be in `~/.config/kargo/` on a Mac when `XDG_CONFIG_HOME` is undefined, you'll be surprised. It's in `~/Library/Application Support/kargo` instead. Surprising though it may be, it's arguably correct.

Just let me know if you want to keep it this way or if you want me to do what Argo CD's CLI does and use `~/.config/kargo/` if `XDG_CONFIG_HOME` is undefined.

Personally, I have no preference. Outside of Linux, there doesn't seem to be an obvious _right_ way of doing this.

Reply:
Works for me.

Reply:
@jessesuen -- fixed here : https://github.com/akuity/kargo/pull/508/files#diff-8059bb3610626754760c9cf8f25d8b8060ad75f6b9634e33e01c6db81a32caf5R14-R35